eclipse/src/saros/project/ProjectDeltaVisitor.java

```diff
@@ -170,7 +164,18 @@ private void handleFileDelta(IResourceDelta delta) {
 
   private void generateCreated(IResource resource) {
 
-    saros.filesystem.IResource wrappedResource = ResourceAdapterFactory.create(resource);
+    saros.filesystem.IResource wrappedResource =
+        ResourceConverter.convertToResource(referencePoint, resource);
+
+    if (wrappedResource == null) {
+      log.error(
```

Comment:
I am not quite sure what you mean by "unsharing".

If you mean "removing a reference point from the session": I am not sure whether reducing the resource scope was ever considered for the session design. This would probably also lead to a lot of other issues.

If you mean "excluding specific child resources of a reference point from the session": Not necessarily. The current logic operates solely on paths, so it would still return resources on such ignored/excluded children. This is, however, missing a check to not send activities for such files. I will add an `isShared` check.

Reply:
Well there is a referencePointRemove method, but yeah you are right. If you really try to implement this things will get worse and you will find a lot of design issues / choice that will render that to almost unable to implement it properly.